### PR TITLE
feat: Cache SQL

### PR DESCRIPTION
**Pull request by kgilpin**
## Features

**Performance**

Reduces the time to scan `appmap-server`.

Without SQL cache:

```
Performed 7860 checks in 22231ms (353 checks/sec)
Parse count: 7708
Normalize count: 8279
yarn start scan -c ../appland/appland-scanner.yml -d ../appland/tmp/appmap     29.35s user 1.25s system 117% cpu 26.010 total
```

With SQL cache:

```
Performed 7860 checks in 7142ms (1100 checks/sec)
Parse count: 225
Normalize count: 1331
yarn start scan -c ../appland/appland-scanner.yml -d ../appland/tmp/appmap     13.88s user 0.87s system 133% cpu 11.047 total
```

**Output**

Writes SQL parse errors to file _sql_warning.txt_.

Reports the elapsed time and the number of checks/sec.

### CODE REVIEW

File: src/database/index.ts [hunk 1]
```diff
@@ -4,8 +4,9 @@
 // TODO: Copied and TypeScript-ified from @appland/models
 
 import { visit } from './visit';
-import { Event, getSqlLabelFromString, SqlQuery } from '@appland/models';
+import { Event, /* getSqlLabelFromString, */ SqlQuery } from '@appland/models';
```
Comment (dustinbyrne) [inline src/database/index.ts]: This commit is still named `wip`. Might as well drop this comment too.

File: sql_warning.txt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
...":"call","defined_class":"SessionController","method_id":"create","path":"app/controllers/session_controller.rb","lineno":4,"receiver":{"class":"SessionController","value":"#<SessionController:0x0000000000000000>","object_id":"0"},"http_server_request":{"request_method":"POST","normalized_path_info":"/sessions/new","path_info":"/sessions/new","client_normalized_path_info":"/sessions/new"},"message":[{"name":"login","class":"String","object_id":70252828231460,"value":"alice"}]},{"id":2,"event":"call","defined_class":"Session","method_id":"initialize","path":"app/models/session.rb","lineno":4,"receiver":{"class":"Session","value":"#<Session:0x0000000000000000>","object_id":"1"},"sql_query":{"sql":"SELECT * FROM users WHERE login = ''alice''","normalized_sql":"SELECT * FROM users WHERE login = $1","normalized":true}},{"id":3,"event":"return","defined_class":"Session","method_id":"initialize","path":"app/models/session.rb","lineno":4,"parent_id":2},{"id":4,"event":"return","defined_class":"SessionController","method_id":"create","path":"app/controllers/session_controller.rb","lineno":4,"parent_id":1}]}'::jsonb, "sha_256" = '\x... : Syntax error found near Column Identifier (SET Clause)
```
Comment (dustinbyrne) [inline sql_warning.txt]: Looks like something happened with this commit. Should these files be added? I don't see any changes to the finding hash.

**Comment by dustinbyrne** [comment]
Looks like something happened to this commit too? 7665c36e92d1347f1307093d764dda5a5fe6ccfa

File: sql_warning.txt (hunk 2 above)
Comment (kgilpin) [reply sql_warning.txt]: Yeah, this happened during rebase for some reason. I've dropped the whole commit, and the 7665c36 commit as well.

**Review by dustinbyrne — APPROVED** [review]

**Pull request merged**